src/helper/remote_config/protocol/tuf/parser.cpp

```diff
@@ -0,0 +1,392 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog
+// (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
+
+#include <iostream>
+#include <rapidjson/document.h>
+#include <rapidjson/prettywriter.h>
+
+#include "parser.hpp"
+#include <base64.h>
+
+namespace dds::remote_config::protocol {
+
+remote_config_parser_result validate_field_is_present(
+    const rapidjson::Value &parent_field, const char *key, rapidjson::Type type,
+    rapidjson::Value::ConstMemberIterator &output_itr,
+    const remote_config_parser_result &missing,
```

Comment:
Since these are enums, passing references will not be more efficient than passing by value.

Reply:
Done